### PR TITLE
2024 April Update: Section 1.2.x Logic, Section 19 HKU Improvement, Handler Fixes, Prelim Fixes

### PR DESCRIPTION
**Pull request by frederickw082922**
Overall Review of Changes:

Thank you @MrSteve81 for the enhancements to this release!

- Improved 19.x section logic for Windows local user SIDs and HKU support.
- Reboot handler and logic Improvement with skip_reboot var feature.
- win_skip_for_test var update with additional description
- Mislabeled control fix for win19cis_rule_2_3_1_4, win19cis_rule_18_9_7_2, win19cis_rule_18_9_32_6_2, win19cis_rule_2_3_1_1, win19cis_rule_2_3_1_2, win19cis_rule_2_3_1_3
- Improved logic for win19cis_cloud_based_system 1.2.x controls.

Issue Fixes:
Azure Cloud Logic for 1.2.x Section from @mfortin

Enhancements:
Improved Reboot Handler and SID user Logic for all local accounts

How has this been tested?:
Locally and Azure

### CODE REVIEW

File: tasks/prelim.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@
 # Current list is elastic and will be updated as we test more cloud based services.
 # Current testing is working in Azure using Hyper-V. We are currently using this for reference:
 # https://github.com/ansible/ansible/blob/905131fc76a07cf89dbc8d33e7a4910da3f10a16/lib/ansible/module_utils/facts/virtual/linux.py#L205
-- name: Set Fact If Cloud-Based System.
+- name: PRELIM | Set Fact If Cloud-Based System.
   ansible.builtin.set_fact:
       win19cis_cloud_based_system: true
   when:
```
Comment (mfortin) [inline tasks/prelim.yml:32]: wouldn't the Azure fix be here ?

Comment (mfortin) [inline tasks/prelim.yml:32]:
```suggestion
  when:
      - ansible_system_vendor == 'Microsoft Corporation'
```

Comment (frederickw082922) [reply tasks/prelim.yml:32]: Oversight! Thank you @mfortin ! I see you put in your PR #99 ! Nice work!

**Review by georgenalen — APPROVED** [review]
Looks good!

**Comment by frederickw082922** [comment]
Thank you @georgenalen !

**Pull request merged**